src/click/testing.py

```diff
@@ -380,18 +380,25 @@ def invoke(
         )
 
     @contextlib.contextmanager
-    def isolated_filesystem(self):
+    def isolated_filesystem(self, tempdir=None):
```

Comment:
So presumably, you can add a test [here](https://github.com/pallets/click/blob/master/tests/test_testing.py) to exercise this new behavior. 

Actually, since this function is not tested directly (at present), you might want to add two tests: one where you call `isolated_filesystem` with an arg, and one without.

In each test, you can just use Python's `os` library to inspect the temporary directory. You should be able to call `os.path.exists` on the yielded directory both during and after the lifetime of the context manager. Something like

```
lingering_directory = None
with CLIRunner.isolated_filesystem() as directory:
    lingering_directory = directory
    self.assertTrue(os.path.exists(directory))
self.assertFalse(os.path.exists(directory))
```

I leave it to you to determine the testing logic for when `isolated_filesystem` takes an argument.